platform/android/java_godot_lib_jni.cpp

```diff
@@ -163,6 +165,8 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_setup(JNIEnv *env, jc
 }
 
 JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_resize(JNIEnv *env, jclass clazz, jobject p_surface, jint p_width, jint p_height) {
+	std::lock_guard<std::recursive_mutex> lock(step_resize_lock);
```

Comment:
What's the use of the mutex here for?

Reply:
Mutex guards `Java_org_godotengine_godot_GodotLib_step()` and `Java_org_godotengine_godot_GodotLib_resize()` functions. This is because the former is called on a different thread than the latter function. If there was no mutex, `Java_org_godotengine_godot_GodotLib_resize()` would destroy framebuffers which are currently being used.

Reply:
A more consistent approach would be to add a `surfaceChanged` flag in `VkThread` which is reset every time `onSurfaceChanged` is invoked.
Then in `VkThread::run()` method, move the `vkRenderer.onVkSurfaceChanged(vkSurfaceView.holder.surface, width, height)` call outside of the `if (!rendererResumed) {` block into its own block:

```
if (surfaceChanged) {
   vkRenderer.onVkSurfaceChanged(vkSurfaceView.holder.surface, width, height);
surfaceChanged = false;
}
```

This will allow you to get rid of the mutex since `resize` and `step` would always be invoked on the same thread.

Reply:
I like this idea, thanks. Fixed